include/powsybl/iidm/ExtensionAdder.hpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef POWSYBL_IIDM_EXTENSIONADDER_HPP
+#define POWSYBL_IIDM_EXTENSIONADDER_HPP
+
+#include <memory>
+
+namespace powsybl {
+
+namespace iidm {
+
+class Extendable;
+class Extension;
+
+class ExtensionAdder {
+public:
+    /**
+     * Copy assignment operator
+     */
+    ExtensionAdder& operator=(const ExtensionAdder&) = delete;
+
+    /**
+     * Move assignment operator
+     */
+    ExtensionAdder& operator=(ExtensionAdder&&) noexcept = delete;
+
+    /**
+     * Add the extension.
+     *
+     * @return the extendable
+     */
+    Extendable& add();
+
+protected:
+    /**
+     * Constructor
+     */
+    explicit ExtensionAdder(Extendable& extendable);
+
+    /**
+     * Destructor
+     */
+    virtual ~ExtensionAdder() noexcept = default;
+
+    /**
+     * Copy constructor
+     */
+    ExtensionAdder(const ExtensionAdder& extendable) = default;
+
+    /**
+     * Move constructor
+     */
+    ExtensionAdder(ExtensionAdder&&) noexcept = default;
```

Comment:
If you make the constructor movable and copyable, you should probably do the same with `operator=`